libs2eplugins/src/s2e/Plugins/Models/BaseFunctionModels.cpp

```diff
@@ -81,16 +82,21 @@ bool BaseFunctionModels::findNullCharWithWidth(S2EExecutionState *state, uint64_
 
         bool truth;
         bool res = solver->mustBeTrue(query, truth);
-        if (res && truth) {
+
+        if (truth&&res) {
+            allSymbolic = false; // Found a null byte, not all characters are symbolic
+            getDebugStream(state) << "Found nullptr at offset " << len << "\n";
             break;
         }
     }
 
-    if (len == MAX_STRLEN) {
-        getDebugStream(state) << "Could not find nullptr char\n";
-        return false;
+    if (allSymbolic && len == MAX_STRLEN) {
```

Comment:
It can be all chars are concrete but all of them are not null char.

Reply:
yes. updated